tests/testthat/test-Logger.R

```diff
@@ -1,240 +1,468 @@
 
-  utils::capture.output(logger$log_info("test message"))
+  # Set options for the test
+  log_path <- tempdir(check = TRUE)
+  db_table <- "test.SCDB_logger"
+  timestamp <- Sys.time()
+
+  # Create logger1 and logger2 which uses the same file
+  # Since start_time is the same for both
+  logger1 <- Logger$new(
+    db_table = db_table,
+    timestamp = timestamp,
+    log_path = log_path,
+    output_to_console = FALSE
+  )
 
   logger2 <- Logger$new(
-    db_tablestring = db_tablestring,
-    ts = ts,
-    start_time = ts,
-    log_table_id = NULL,
-    log_path = log_path
+    db_table = db_table,
+    timestamp = timestamp,
+    start_time = timestamp,
+    log_path = log_path,
+    output_to_console = FALSE
   )
 
+
+  # logger1 should be able to successfully write
+  logger1$log_info("test message")
+
+
+  # whereas logger2 should fail since the log file now exists
   expect_error(
-    logger2$log_info("test message")
+    logger2$log_info("test message"),
+    "already exists!"
   )
 
-  file.remove(file.path(log_path, logger$log_filename))
+  file.remove(logger1$log_realpath)
 
-  rm(logger, logger2)
-  gc()
+  rm(logger1, logger2)
+  invisible(gc())
 })
 
-test_that("Logger console output may be suppressed", {
-  db_tablestring <- "test.SCDB_logger"
-  ts <- Sys.time()
-  log_path <- tempdir()
+
+test_that("Logger: console output may be disabled", {
 
   # First test cases with output_to_console == FALSE
   # Here, only print when explicitly stated
-  logger <- Logger$new(
-    db_tablestring = db_tablestring,
-    ts = ts,
-    log_table_id = NULL,
-    log_path = log_path,
-    output_to_console = FALSE
+  expect_warning(
+    logger <- Logger$new(output_to_console = FALSE),
+    regexp = "NO LOGGING WILL BE DONE"
   )
 
-  # In lieu of testthat::expect_no_output...
-  expect_identical(
-    utils::capture.output(logger$log_info("Whoops! This should not have been printed!"), type = "output"),
-    character(0)
-  )
-  expect_identical(
-    utils::capture.output(logger$log_info("Whoops! This should not have been printed either!",
-                                          output_to_console = FALSE),
-                          type = "output"),
-    character(0)
-  )
+  expect_no_message(logger$log_info("Whoops! This should not have been printed!"))
+
+  expect_no_message(logger$log_info("Whoops! This should not have been printed either!", output_to_console = FALSE))
 
-  expect_output(
-    logger$log_info("This line should be printed",
-                    output_to_console = TRUE),
-    regex = "This line should be printed$"
+  expect_message(
+    logger$log_info("This line should be printed", output_to_console = TRUE),
+    "This line should be printed"
   )
 
+  rm(logger)
+
   # ...and now, only suppress printing when explicitly stated
-  logger$output_to_console <- TRUE
+  expect_warning(
+    logger <- Logger$new(output_to_console = TRUE),
+    regexp = "NO LOGGING WILL BE DONE"
```

Comment:
Maybe the function should be changed so that it states that logging is only done to the console?